tests/pydantic/test_primitive_list.py

```diff
@@ -0,0 +1,166 @@
+"""Test special cases for extension of JSON-primitive fields."""
+import pytest
+import typing
+
+from fhirzeug.generators.python_pydantic.templates.resource_header import (
+    FHIRAbstractBase,
+    get_primitive_field_root_validator,
+)
+import pydantic
+
+OPTIONAL_LIST_T = typing.Optional[typing.List[typing.Optional[str]]]
+
+
+class ExampleModel(FHIRAbstractBase):
+    """Model for tests that simulate the behavior of a generated resource."""
+
+    field: OPTIONAL_LIST_T
+    field__extension: OPTIONAL_LIST_T
+
+    # Encountered once an issue with field names using "snake_case"
+    # It is now tested for robustness.
+    snake_field: OPTIONAL_LIST_T
+    snake_field__extension: OPTIONAL_LIST_T
+
+    _validate_primitive_field = get_primitive_field_root_validator("field")
+    _validate_primitive_snake_field = get_primitive_field_root_validator("snake_field")
+
+
+class ContainerModel(FHIRAbstractBase):
+    """Container model to test validation on submodels."""
+
+    example: ExampleModel
+
+
+@pytest.mark.parametrize(
+    (
+        "field_value",
+        "field_extension_value",
+        "field_validated_value",
+        "field_extension_validated_value",
+    ),
+    [
+        (["A"], ["extA"], ["A"], ["extA"]),
+        (["A", None], None, ["A"], None),
+        (["A", "B"], None, ["A", "B"], None),
+        (["A", None], [None, "extB"], ["A", None], [None, "extB"]),
+        (["A", "B"], [None, None], ["A", "B"], [None, None]),
+        (["A", "B"], [], ["A", "B"], None),  # Valid but empty list is set to None
+        (["A", "B"], [None], ["A", "B"], None),  # Valid but None list is set to None
```

Comment:
From what has been decided and implemented, this example would fail because the last element of both lists are `None`. At least 1 of them must be filled.